src/test/kotlin/no/nav/helsearbeidsgiver/innsending/InnsendingRouteTest.kt

```diff
@@ -0,0 +1,182 @@
+import io.ktor.http.contentType
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockkStatic
+import io.mockk.unmockkStatic
+import io.mockk.verify
+import kotlinx.coroutines.test.runTest
+import no.nav.helsearbeidsgiver.authorization.ApiTest
+import no.nav.helsearbeidsgiver.config.Services
+import no.nav.helsearbeidsgiver.domene.inntektsmelding.v1.AarsakInnsending
+import no.nav.helsearbeidsgiver.inntektsmelding.Arbeidsgiver
+import no.nav.helsearbeidsgiver.inntektsmelding.InntektsmeldingRequest
+import no.nav.helsearbeidsgiver.utils.DEFAULT_ORG
+import no.nav.helsearbeidsgiver.utils.gyldigSystembrukerAuthToken
+import no.nav.helsearbeidsgiver.utils.json.toJson
+import no.nav.helsearbeidsgiver.utils.mockForespoersel
+import no.nav.helsearbeidsgiver.utils.mockInntektsmeldingRequest
+import no.nav.helsearbeidsgiver.utils.mockInntektsmeldingResponse
+import no.nav.helsearbeidsgiver.utils.opprettImTransaction
+import no.nav.helsearbeidsgiver.utils.test.wrapper.genererGyldig
+import no.nav.helsearbeidsgiver.utils.wrapper.Orgnr
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.util.UUID
+
+class InnsendingRouteTest : ApiTest() {
+    @BeforeEach
+    fun setup() {
+        mockkStatic(Services::opprettImTransaction)
+        every { services.opprettImTransaction(any(), any()) } just Runs
+    }
+
+    @Test
+    fun `innsending av inntektsmelding på gyldig forespørsel`() =
+        runTest {
+            val requestBody = InnsendingMockData.requestBody
+            val forespoersel = InnsendingMockData.forespoersel
+            every { repositories.forespoerselRepository.hentForespoersel(forespoersel.navReferanseId) } returns forespoersel
+            every { repositories.inntektsmeldingRepository.hent(forespoersel.navReferanseId) } returns emptyList()
+            val response = sendInnInntektsmelding(requestBody)
+            response.status shouldBe HttpStatusCode.Created
+            verify {
+                services.opprettImTransaction(
+                    match { it.type.id == requestBody.navReferanseId },
+                    match { it.type.id == requestBody.navReferanseId },
+                )
+            }
+        }
+
+    @Test
+    fun `innsending av inntektsmelding uten gyldig forespørsel gir bad request`() =
+        runTest {
+            val requestBody = InnsendingMockData.requestBody.copy(navReferanseId = UUID.randomUUID())
+            every { repositories.forespoerselRepository.hentForespoersel(requestBody.navReferanseId) } returns null
+            val response = sendInnInntektsmelding(requestBody)
+            response.status shouldBe HttpStatusCode.BadRequest
+            verify(exactly = 0) {
+                services.opprettImTransaction(
+                    match { it.type.id == requestBody.navReferanseId },
+                    match { it.type.id == requestBody.navReferanseId },
```

Comment:
På alle de testene her som har `verify(exactly = 0)`, trenger vi egentlig matcherne her? Er det ikke enda mer robust å sjekke at `services.opprettImTransaction()` aldri blir kalt med noen som helst argumenter?
```suggestion
                    any(),
                    any(),
```